experimental/packages/opentelemetry-instrumentation/src/instrumentation.ts

```diff
@@ -61,7 +61,28 @@ export abstract class InstrumentationAbstract<T = any>
   }
 
   /* Api to wrap instrumented method */
-  protected _wrap = shimmer.wrap;
+  protected _wrap = (
+    moduleExports: any,
+    name: string,
+    wrapper: (originalFn: any) => any
+  ) => {
+    try {
+      return shimmer.wrap(moduleExports, name, wrapper);
+    } catch (e) {
+      // shimmer doesn't handle Proxy objects well
```

Comment:
couldn't we use `instanceof Proxy` instead of letting it throw an error ?

Reply:
This has now been updated to detect Proxy with `util.types.isProxy(moduleExports)`... and there is some new splitting of wrapping functionality. Instead of wrapping from within the shared `instrumentation.ts` file, this is being handled in the node-specific `instrumentation.ts`. When a Proxy is detected, it uses a new `_wrapEsm` function to properly wrap the Proxy object.